### PR TITLE
Update session validators for Laravel 5.

### PR DESCRIPTION
**Pull request by DFurnes**
# Changes
 - Update `UserSessionRequest` and `AdminSessionRequest` validation to work with Laravel 5 [Form Request](http://laravel.com/docs/master/validation#form-request-validation) objects. Registration validation uses Controller Validation, because it seems that Form Requests can only be used through dependency injection on route-handling controller methods.
 - Remove "Forgot Password" link because `Route::controller` seems to be causing issues in L5.

For review: @angaither

### CODE REVIEW

**Comment by DFurnes** [comment]
Also cleaned up routes, so instead of having the kinda confusing `session/*` _and_ named (`login`, etc.) routes, there are only `login`, `admin`, and `logout` routes which map directly to methods on SessionsController.

File: app/Http/Controllers/SessionsController.php [hunk 1]
```diff
@@ -119,11 +101,11 @@ public function userLogin()
...
     /**
      * Authentication for an admin user.
-     * @see SessionsController#create()
+     * @param AdminSessionRequest $request
      */
-    public function adminLogin()
+    public function adminLogin(AdminSessionRequest $request)
     {
-        $credentials = Input::only('email', 'password');
+        $credentials = $request->only('email', 'password');
```
Comment (angaither) [inline app/Http/Controllers/SessionsController.php]: huh, I thought he hated `$` strange that this changed in L5

File: app/Http/Requests/UserSessionRequest.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public function messages()
+    {
+        return [
+            'first_name.required' => 'What\'s your name?!',
+            'phone.required_without' => 'Give us your digits!',
+            'phone.regex' => 'That doesn\'t look like a real phone number!',
+            'email.required_without' => 'We need your email',
+            'email.email' => 'We need a valid email',
+            'birthdate.required' => 'When were you born?',
+            'birthdate.date' => 'Enter your birthday MM/DD/YYYY!',
```
Comment (angaither) [inline app/Http/Requests/UserSessionRequest.php]: do you need a rule for `before:today` ?

**Comment by angaither** [comment]
:ok_hand:.

File: app/Http/Requests/UserSessionRequest.php (hunk 2 above)
Comment (DFurnes) [inline app/Http/Requests/UserSessionRequest.php]: Yep, it's in the `$rules` variable a few lines up. Not sure what the default error message is, but it'll use that unless we override here.

**Pull request merged**